.circleci/config.yml

```diff
@@ -53,7 +53,14 @@ commands:
           at: /tmp
       - run:
           name: 'Maybe Gracefully Halt'
-          command: if ls /tmp/workspace/.CI_GRACEFULLY_HALT_* 1>/dev/null 2>&1; then echo "Gracefully halting this job."; circleci-agent step halt; fi
+          command: |
+            if [[ -d /tmp/workspace ]]; then
+              mv /tmp/workspace /tmp/restored-workspace
```

Comment:
From my understanding of this code, it will take effect when an additional layer is added to our CI matrix. Are there any plans to do this? (Or is there something else at play?)

Reply:
Yes, I said that in the PR description :)

Reply:
I read the description. I was asking about what the upcoming change is that requires a split in the workspace. The main concern being that adding a third (fourth?) stage to CI has the potential to reduce overall throughput for the organization due to increased VM use.

Reply:
wdym by "stage"? Are you talking about more steps inside a job, or another "layer" of jobs inside the workflow graph?

If it's the former, steps are cheap - they're just a way to group commands and outputs, but they run on the same shell instance.

If it's the latter, then that's not the thing I'm planning on changing. The reason I wanted to split the restored and the new workspace for each job is so that devs working on pr-check/ files (i.e., wg-infra) don't have to worry about micro-managing the workspace between the jobs. Incoming files are always going to be in /tmp/restored-workspace, outgoing files go in /tmp/workspace - and you won't have to worry about leaving existing files there and causing a workspace merge conflict

Reply:
> If it's the latter, then that's not the thing I'm planning on changing. The reason I wanted to split the restored and the new workspace for each job is so that devs working on pr-check/ files (i.e., wg-infra) don't have to worry about micro-managing the workspace between the jobs. Incoming files are always going to be in /tmp/restored-workspace, outgoing files go in /tmp/workspace - and you won't have to worry about leaving existing files there and causing a workspace merge conflict

Yes, my concern was indeed the latter, and this explanation is exactly what I was looking for. Thanks for indulging me, carry on!